File: iOS/PanoAudioChat/PanoAudioChatSDK/Source/Services/PanoSignalService.m
//
//  PanoSignalService.m
//  PanoAudioChatSDK
//
//  Created by pano on 2021/7/15.
//

#import "PanoSignalService.h"
#import "PanoClientService+OpenApi.h"


@implementation PanoSignalService

- (id <PanoSignalInterface>)signalService {
    return PanoClientService.service;
}

- (BOOL)sendInviteMsg:(PanoMicInfo *)micInfo {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:inviteUser data:@[micInfo] reason:ok] toDictionary];
    return [self.signalService sendMessage:params toUser:micInfo.userId];
}

- (BOOL)sendAcceptInviteMsg:(PanoMicInfo *)micInfo {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:acceptInvite data:@[micInfo] reason:ok] toDictionary];
    return [self.signalService sendMessage:params toUser:PanoClientService.service.hostUserId];
}

- (BOOL)sendRejectInviteMsg:(PanoMicInfo*)micInfo
                     reason:(PanoCmdReason)reason {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:rejectInvite data:@[micInfo] reason:reason] toDictionary];
    return [self.signalService sendMessage:params toUser:PanoClientService.service.hostUserId];
}

- (BOOL)sendkillMsg:(PanoMicInfo *)micInfo {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:killUser data:@[micInfo] reason:ok] toDictionary];
    return [self.signalService sendMessage:params toUser:micInfo.userId];
}

- (BOOL)sendApplyChatMsg:(PanoMicInfo *)micInfo {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:applyChat data:@[micInfo] reason:ok] toDictionary];
    return [self.signalService sendMessage:params toUser:PanoClientService.service.hostUserId];
}

- (BOOL)sendCancelChatMsg:(PanoMicInfo *)micInfo {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:cancelChat data:@[micInfo] reason:ok] toDictionary];
    return [self.signalService sendMessage:params toUser:PanoClientService.service.hostUserId];
}

- (BOOL)sendAcceptChatMsg:(PanoMicInfo *)micInfo {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:acceptChat data:@[micInfo] reason:ok] toDictionary];
    return [self.signalService sendMessage:params toUser:micInfo.userId];
}

- (BOOL)sendRejectChatMsg:(PanoMicInfo*)micInfo
                   reason:(PanoCmdReason)reason {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:rejectChat data:@[micInfo] reason:reason] toDictionary];
    return [self.signalService sendMessage:params toUser:micInfo.userId];
}

- (BOOL)sendStopChatMsg:(PanoMicInfo *)micInfo {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:stopChat data:@[micInfo] reason:ok] toDictionary];
    return [self.signalService sendMessage:params toUser:PanoClientService.service.hostUserId];
}

- (BOOL)sendCloseRoomMsg {
    NSDictionary *params = [[[PanoMicMessage alloc] initWithCmd:closeRoom data:nil reason:ok] toDictionary];
    return [self.signalService broadcastMessage:params];
}

@end

File: iOS/PanoAudioChat/PanoAudioChat/Classes/Service/PanoSignalService.swift
//
//  PanoCmdMsgService.swift
//  PanoAudioChat
//
//  Created by wenhua.zhang on 2021/6/11.
//

import Foundation


class PanoSignalService {
    
    var signalService: PanoSignalInterface {
        get {
            return PanoServiceManager.service(type: .client) as! PanoSignalInterface;
        }
    }
    
    func sendInviteMsg(micInfo: PanoMicInfo) -> Bool {
        let params = PanoMicMessage(cmd: .inviteUser, data: [micInfo]).toDictionary()
        return self.signalService.sendMessage(msg: params, toUser:  micInfo.userId)
    }
    
    func sendAcceptInviteMsg(micInfo: PanoMicInfo) -> Bool {
        let params = PanoMicMessage(cmd: .acceptInvite, data: [micInfo]).toDictionary()
        return self.signalService.sendMessage(msg: params, toUser:  PanoClientService.hostUserId)
    }
    
    func sendRejectInviteMsg(micInfo: PanoMicInfo, reason: PanoCmdReason) -> Bool {
        let params = PanoMicMessage(cmd: .rejectInvite, data: [micInfo], reason: reason).toDictionary()
        return self.signalService.sendMessage(msg: params, toUser:  PanoClientService.hostUserId)
    }
    
    func sendkillMsg(micInfo: PanoMicInfo) -> Bool {
        let params = PanoMicMessage(cmd: .killUser, data: [micInfo]).toDictionary()
        return self.signalService.sendMessage(msg: params, toUser:  micInfo.userId)
    }
    
    func sendApplyChatMsg(micInfo: PanoMicInfo) -> Bool {
        let params = PanoMicMessage(cmd: .applyChat, data: [micInfo]).toDictionary()
        return self.signalService.sendMessage(msg: params, toUser:  PanoClientService.hostUserId)
    }
    
    func sendCancelChatMsg(micInfo: PanoMicInfo) -> Bool {
        let params = PanoMicMessage(cmd: .cancelChat, data: [micInfo]).toDictionary()
        return self.signalService.sendMessage(msg: params, toUser:  PanoClientService.hostUserId)
    }
    
    func sendAcceptChatMsg(micInfo: PanoMicInfo) -> Bool {
        let params = PanoMicMessage(cmd: .acceptChat, data: [micInfo]).toDictionary()
        return self.signalService.sendMessage(msg: params, toUser: micInfo.userId)
    }
    
    func sendRejectChatMsg(micInfo: PanoMicInfo, reason: PanoCmdReason = .ok) -> Bool {
        let params = PanoMicMessage(cmd: .rejectChat, data: [micInfo], reason: reason).toDictionary()
        return self.signalService.sendMessage(msg: params, toUser:  micInfo.userId)
    }
    
    func sendStopChatMsg(micInfo: PanoMicInfo) -> Bool {
        let params = PanoMicMessage(cmd: .stopChat, data: [micInfo]).toDictionary()
        return self.signalService.sendMessage(msg: params, toUser:  PanoClientService.hostUserId)
    }
    
    func sendCloseRoomMsg() -> Bool {
        return self.signalService.broadcastMessage(msg: PanoMicMessage(cmd: .closeRoom).toDictionary())
    }
}
